go-controller/pkg/cni/cni.go

```diff
@@ -180,23 +180,26 @@ func (pr *PodRequest) getCNIResult(podInterfaceInfo *PodInterfaceInfo) (*current
 	}, nil
 }
 
-// getAnnotationsOnPod obtains the pod annotation from the cache
-func getAnnotationsOnPod(podLister corev1listers.PodLister, namespace, name string) (map[string]string, error) {
-	var annotations map[string]string
-
-	if err := wait.PollImmediate(200*time.Millisecond, 30*time.Second, func() (bool, error) {
-		pod, err := podLister.Pods(namespace).Get(name)
-		if err != nil {
-			return false, err
-		}
-		annotations = pod.ObjectMeta.Annotations
-		if _, ok := annotations[util.OvnPodAnnotationName]; ok {
-			return true, nil
+// getPodAnnotations obtains the pod annotation from the cache
+func getPodAnnotations(ctx context.Context, podLister corev1listers.PodLister, namespace, name string) (map[string]string, error) {
+	timeout := time.After(30 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("canceled waiting for annotations")
+		case <-timeout:
+			return nil, fmt.Errorf("timed out waiting for annotations")
+		default:
+			pod, err := podLister.Pods(namespace).Get(name)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get annotations: %v", err)
```

Comment:
maybe a better error message here like "unable to find pod (probably deleted)" or something

Reply:
@trozet since we're asking the informer cache now, rather than the API, we'll get a descriptive error from:

```
func (s podNamespaceLister) Get(name string) (*v1.Pod, error) {
	obj, exists, err := s.indexer.GetByKey(s.namespace + "/" + name)
	if err != nil {
		return nil, err
	}
	if !exists {
		return nil, errors.NewNotFound(v1.Resource("pod"), name)
	}
	return obj.(*v1.Pod), nil
}
```